architecture/adr-list/adr-007-estimating-transaction-fees.md

```diff
@@ -0,0 +1,133 @@
+
+The `gas-wanted` estimation functionality can also be accessed via RPC calls by querying the `app/simulate` endpoint.
+
+#### **`gas-used`**
+
+`gas-used` is the amount of gas used by a transaction and is always less than  `gas-wanted`.
+
+If a transaction requires more gas than specified in `gas-wanted`, execution is interrupted. The fee is NOT returned and becomes fully exhausted.
+
+#### **`gas-prices`**
+
+`gas-prices` is the price of gas suggested by transaction sender. It is set by the transaction sender and can be set to any value.
+
+Validators can define a parameter called `min-gas-prices`, which acts as a filter on the transactions in each validators' mempool. Transactions with lower `gas-prices` are not considered as candidates for inclusion into the next blocks by the validator. This value is configured by the node/validator operator in the `app.toml` config file.
+
+If `gas-prices` specified in a transaction is lower than the `min-gas-prices` of _all_ validators, no validators consider that transaction in their mempool. The transaction eventually times out and is not processed.
+
+If `gas_prices` specified in a transaction is lower than `min-gas-prices` of _some_ validators, the transaction has a chance at being committed within the transaction timeout window. Whether this actually happens depends on the sum of voting powers of validators that are ready to commit it.
+
+There is currently no way to request `min-gas-prices`.
+
+#### **`fee`**
+
+`fee` is the amount of tokens a validator takes for a transaction processing and is calculated as `fee` = `gas` \* `gas-prices`
+
+`gas` above must be at least equal to or more than `gas-wanted`. 
+
+### Options considered for estimating transaction fees
+
+#### Gas estimation
+
+**Option 1: Client-side calculation**
+
+1. A gas estimation request could be built into the client libraries \(such as [VDR Tools](https://gitlab.com/evernym/verity/vdr-tools)\) used in applications using cheqd network
+2. The client library would calculate an adjustment coefficient client-side.
+
+**Option 2: Use estimated values**
+
+1. Estimated gas for most common transactions could be pre-calculated based on best-guess.
+2. This method can be used as a workaround in the absence of a formal gas estimation request in Cosmos.
+3. This scenario is more likely to fail over time as gas prices can be dynamic on a live network with lots of transactions.
+
+#### Fee price estimations
+
+**Option 1: Use fixed values**
+
+1. Set a fixed recommended gas price on the cheqd network
+2. Embed the fixed value into applications
+
+**Option 2: Dynamically calculate values**
+
+1. Gas prices can be dynamically determined client side based on recent transactions.
+2. This could be implemented as a client-side calculation, provided as a service \(e.g., via an oracle\), or a blend of both approaches.
```

Comment:
Same as above, potentially a weighted moving average?

Reply:
Yep, moving average is a great idea